graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -350,18 +357,136 @@ def run(self) -> None:
+    ) -> dict:
+        """Merge the config dict with the transformations dict and return a copy.
+
+        Parameters
+        ----------
+        gsp_config_dict : dict
+            The input configuration dictionary, using GSProcessing schema
+        transformations : Mapping[str, Mapping[str, Mapping]]
+            The processed graph representations containing the transformations.
+            Expected dict schema:
+
+            {
+                "node_features": {
+                    "node_type1": {
+                        "feature_name1": {
+                            "transformation": # transformation type
+                            # feature1 representation goes here
+                        },
+                        "feature_name2": {}, ...
+                    },
+                    "node_type2": {...}
+                },
+                "edges_features": {...}
+            }
+
+        Returns
+        -------
+        dict
+            A copy of the ``gsp_config_dict`` modified to be merged with
+            the transformation representations.
+        """
+        gsp_config_dict_copy = copy.deepcopy(gsp_config_dict)
+
+        edge_transformations: Mapping[str, Mapping[str, Mapping]] = transformations["edge_features"]
+        node_transformations: Mapping[str, Mapping[str, Mapping]] = transformations["node_features"]
+        edge_input_dicts: list[dict] = gsp_config_dict_copy["edges"]
+        node_input_dicts: list[dict] = gsp_config_dict_copy["nodes"]
+
+        def get_structure_type(single_input_dict: dict, structure_type: str) -> str:
+            """Gets the node or edge type name from input dict."""
+            if structure_type == "node":
+                type_name = single_input_dict["type"]
+            elif structure_type == "edge":
+                src_type = single_input_dict["source"]["type"]
+                dst_type = single_input_dict["dest"]["type"]
+                relation = single_input_dict["relation"]["type"]
+                type_name = f"{src_type}:{relation}:{dst_type}"
+            else:
+                raise ValueError(
+                    f"Invalid structure type: {structure_type}. Expected 'node' or 'edge'."
+                )
+
+            return type_name
+
+        def append_transformations(
+            structure_input_dicts: list[dict],
+            structure_transformations: Mapping[str, Mapping[str, Mapping]],
+            structure_type: str,
+        ):
+            """Appends the pre-computed transformations to the input dicts."""
+            assert structure_type in ["edge", "node"]
+            for input_dict in structure_input_dicts:
+                type_name = get_structure_type(input_dict, structure_type)
+                # If we have pre-computed transformations for this type
+                if type_name in structure_transformations:
+                    type_transformations: Mapping[str, Mapping] = structure_transformations[
```

Comment:
type_transformations is feature_name -> transformation_information?
Will all the transformation information be a Mapping?

Reply:
Correct, each transformation is `"feature_name" -> dict`. We use Mapping instead of dict because it helps with type annotation, and communicates that we don't expect to modify the mapping. In cases where we expect to modify the contents we use `dict`.

Reply:
Can you add some comments about the "feature_name" -> Mapping ?
For example ` "feature_name" is the feature name and the Mapping records the transformation related meta information such as xxx`